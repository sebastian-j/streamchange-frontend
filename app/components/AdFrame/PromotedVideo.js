import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    top: 33vh;
  }
  50% {
    top: 10vh;
  }
  100% {
    top: 33vh;
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
`;

const PromotedContent = styled.div`
  border: 1px solid #cecece;
  color: ${(props) => props.theme.staticTextColor};
  display: flex;
  flex-direction: row;
  padding: 14px;
`;

const PromotedVideo = (props) => (
  <PromotedContentWrapper>
    <AdTitle>
      <FormattedMessage {...messages.title} />
    </AdTitle>
    <PromotedContent>
      <iframe
        id="ytplayer"
        width="500"
        height="170"
        src={`https://www.youtube.com/embed/${props.videoId}?autoplay=0origin=https://www.streamchange.pl`}
        frameBorder="0"
        title="video"
      />
      <div>{props.description}</div>
    </PromotedContent>
  </PromotedContentWrapper>
);

PromotedVideo.propTypes = {
  description: PropTypes.string,
  videoId: PropTypes.string.isRequired,
};

export default PromotedVideo;
