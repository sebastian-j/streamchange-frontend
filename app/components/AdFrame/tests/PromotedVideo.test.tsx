import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import PromotedVideo from '../PromotedVideo';

const shallowRenderer = createRenderer();

describe('<PromotedVideo />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<PromotedVideo videoId="jNQXAC9IVRw" />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
