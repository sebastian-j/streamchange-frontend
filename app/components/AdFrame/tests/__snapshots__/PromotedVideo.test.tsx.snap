// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromotedVideo /> should render and match the snapshot 1`] = `
<styled.div>
  <styled.span>
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Advertisement"
      id="streamchange.components.AdFrame.title"
    />
  </styled.span>
  <styled.div>
    <iframe
      frameBorder="0"
      height="170"
      id="ytplayer"
      src="https://www.youtube.com/embed/jNQXAC9IVRw?autoplay=0origin=https://www.streamchange.pl"
      title="video"
      width="500"
    />
    <div />
  </styled.div>
</styled.div>
`;
